ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/rest/OkHttpRestClient.java

```diff
@@ -90,16 +97,17 @@ private <T> SafeFuture<Response<T>> getAsyncInternal(
 
   private <T, S> SafeFuture<Response<T>> postAsyncInternal(
       final String apiPath,
+      final Map<String, String> headers,
       final S requestBodyObject,
       final SerializableTypeDefinition<S> requestTypeDefinition,
       final Optional<DeserializableTypeDefinition<T>> responseTypeDefinitionMaybe) {
     final RequestBody requestBody = createRequestBody(requestBodyObject, requestTypeDefinition);
-    final Request request = createPostRequest(apiPath, requestBody);
+    final Request request = createPostRequest(apiPath, requestBody, headers);
     return makeAsyncRequest(request, responseTypeDefinitionMaybe);
   }
 
   private Request createGetRequest(final String apiPath, final Map<String, String> headers) {
-    final HttpUrl httpUrl = createHttpUrl(apiPath);
+    final URL httpUrl = createHttpUrl(apiPath);
```

Comment:
One other small nit, I think returning `HttpUrl` is better, because internally in the library if you pass `URL` it converts it back anyways:

```kotlin
open fun url(url: URL) = url(url.toString().toHttpUrl())
```

Reply:
httpurl was giving an error that it was using an internal function - thats why i chagned it...

Reply:
Ok, fair. Weird one.